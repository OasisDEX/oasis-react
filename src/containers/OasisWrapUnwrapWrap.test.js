/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { fromJS } from 'immutable';

import {
  OasisWrapUnwrapWrapEtherWrapper,
  mapStateToProps,
  mapDispatchToProps
} from './OasisWrapUnwrapWrapEther';
import { shallow } from 'enzyme';

describe('(Container) OasisWrapUnwrapWrap', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps,
    hidden: false
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <OasisWrapUnwrapWrapEtherWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
