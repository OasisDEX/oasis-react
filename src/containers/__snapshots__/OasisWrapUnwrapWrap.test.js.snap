// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) OasisWrapUnwrapWrap should render 1`] = `
<OasisWrapUnwrapWrap
  activeUnwrappedToken="GNT"
  activeUnwrappedTokenBalance={"0"}
  form="wrapEther"
  hidden={false}
  onFormChange={[Function]}
  onSubmit={[Function]}
  transactionState={
    Object {
      "txStartMeta": undefined,
      "txStartTimestamp": undefined,
      "txStatus": undefined,
    }
  }
  txType="TX/GROUP__WRAP_UNWRAP/WRAP"
/>
`;

exports[`(Container) OasisWrapUnwrapWrap will receive right actions 1`] = `
Object {
  "actions": Object {
    "resetActiveWrapForm": [Function],
    "wrapToken": [Function],
  },
}
`;

exports[`(Container) OasisWrapUnwrapWrap will receive right props 1`] = `
Object {
  "activeUnwrappedToken": "GNT",
  "activeUnwrappedTokenBalance": "0",
  "defaultAccount": "0x0000000000000000000000000000000000000000",
}
`;
